lib/HTFeed/JobMetrics.pm

```diff
@@ -308,73 +219,123 @@ sub _valid_metric {
     get_logger()->warn("invalid metric name \"$metric\" at $filename:$line");
     # switch from warn to die to ensure no invalid metric names slip thru:
     # die ("invalid metric name \"$metric\" at $filename:$line");
+
     return 0;
 }
 
+# Labels must be sent as a hashref with simple key-value pairs.
+# Not going to impose any stricter validation at this point.
+# Keys and values will be lowercased.
+# Labels do NOT need to be predefined, unlike metrics.
+sub _valid_labels {
```

Comment:
What happens if we put invalid labels in the output? Does `Prometheus::Tiny::Shared` refuse to use them / raise an error? Or does it put garbage in its output and potentially cause issues down the line of scraping/using those metrics? If the latter, this is definitely worth having. If the former, what extra does this validation routine add?

Reply:
It's mostly training wheels, so I don't start implementing metric names with typoes in them. Prometheus is pretty chill about metric names. I don't trust myself to read, write, or even copy/paste e.g. `validate_zip_completeness_bytes_r_total` consistently, and if you spell it differently in 2 classes you're going to get your data split into 2 measurements.

But I can see that these training wheels are becoming a liability and a limiting factor going forwards.

Reply:
Prometheus is silently discarding any labels it doesn't like, which is why I added this validation.